### PR TITLE
sched: Remove all group id related stuff

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
It is wrong to define a new grpid_t, but not reuse pid_t,
because it make getpid(parent) == getppid(child) impossible.

## Impact
The group id always equals the main task pid

## Testing
Pass ostest

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
Related to https://github.com/apache/incubator-nuttx/pull/2518 and https://github.com/apache/incubator-nuttx/pull/2958.

File: include/nuttx/sched.h [hunk 1]
```diff
@@ -466,14 +460,9 @@ struct task_group_s
 {
 #if defined(HAVE_GROUP_MEMBERS) || defined(CONFIG_ARCH_ADDRENV)
   struct task_group_s *flink;       /* Supports a singly linked list            */
-  grpid_t tg_grpid;                 /* The ID of this task group                */
-#endif
-#ifdef HAVE_GROUP_MEMBERS
-  grpid_t tg_pgrpid;                /* The ID of the parent task group          */
 #endif
-#if !defined(CONFIG_DISABLE_PTHREAD) && defined(CONFIG_SCHED_HAVE_PARENT)
   pid_t tg_task;                    /* The ID of the task within the group      */
```
Comment (patacongo) [inline include/nuttx/sched.h:464]: tg_pid would be a better name.  Then it has the same kind of semantics as tg_ppid.

**Review by patacongo — APPROVED** [review]

**Comment by patacongo** [comment]
Is the MacOS sim failure real?  Appears not.

**Comment by btashton** [comment]
> Is the MacOS sim failure real? Appears not.

It's real scroll up and you can see it in the tcpblaster case. Arm-02 on Linux also failed for the same reason

File: include/nuttx/sched.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/nuttx/sched.h]: Done.

**Comment by xiaoxiang781216** [comment]
Yes, it's my typo error. The build should be OK now.

**Comment by liuguo09** [comment]
> Yes, it's my typo error. The build should be OK now.

Ok, I merge it.

**Pull request merged**